File: 2023/IN/28/codes/Fig_1.py
import matplotlib.pyplot as plt
import numpy as np

def u(n):
    return np.heaviside(n, 1)

x = np.loadtxt("data.dat")
plt.stem(x, u(-x+5)-u(x+3))
plt.title("Plot of x(n) function")
plt.grid(1)
plt.xlabel('n')
plt.ylabel('x(n)')
plt.savefig('fig_1.png')

File: 2023/IN/28/codes/fig_1.c

#include <stdio.h>

int main()
{
    FILE *fp = fopen("data.dat", "w");
    int a = 0;
    for (int i= -8; i<10; i++){
        if(i!=9){
            fprintf(fp, "%d ",i);
            }
        else{
            fprintf(fp, "%d",i);
        }
    }
    fclose(fp);
    

}

